### PR TITLE
chore: replace proteins_pytorch_geometric example with a better one.

### PR DESCRIPTION
**Pull request by ioga**
## Description
Previous example model was flaky and sometimes had "convergence issues", both in our adaptation and in the original code. Since I am no expert on graph convolutional networks, and can't efficiently debug these issues, let's replace the example with a slightly different model which works. Either way, the key point of this example is demoing `get_batch_length` method. 
Also throw in `adaptive.yaml` for good measure.

## Test Plan
It is included in nightlies again.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/graphs/proteins_pytorch_geometric/model_def.py [hunk 1]
```diff
@@ -74,54 +73,43 @@ def __init__(self, context: PyTorchTrialContext):
...
 
         self.optimizer = self.context.wrap_optimizer(
             torch.optim.Adam(
                 self.model.parameters(),
                 lr=self.context.get_hparam("lr"),
-                weight_decay=self.context.get_hparam("weight_decay"),
             )
         )
 
     def train_batch(self, batch, epoch_idx: int, batch_idx: int):
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/model_def.py:93]: Should we add a type annotation here for batch?

File: examples/graphs/proteins_pytorch_geometric/const.yaml [hunk 2]
```diff
@@ -1,19 +1,21 @@
 name: proteins_pytorch_geometric
 hyperparameters:
-  global_batch_size: 20
+  global_batch_size: 60
   dataset: PROTEINS
-  lr: 5.e-4
-  weight_decay: 1.e-4
-  hidden_channels: 32
+  lr: 1.e-3
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/const.yaml:5]: Is there a reason this is different from the original?

File: examples/graphs/proteins_pytorch_geometric/distributed.yaml [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
-    epochs: 300
+    epochs: 200
   smaller_is_better: true
-entrypoint: model_def:GCNTrial
-resources:
-  slots_per_trial: 4
+entrypoint: model_def:GraphConvTrial
 environment:
   image:
     cpu: determinedai/environments:py-3.7-pytorch-1.9-lightning-1.3-tf-2.4-cpu-b06dafb
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/distributed.yaml:21]: Is there a reason we're specifying docker images instead of using the cluster defaults?

File: examples/graphs/proteins_pytorch_geometric/README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
-# PyTorch Geometric - proteins_mincut_pool example
+# PyTorch Geometric - proteins_topk_pool example
 
 This example demonstrates the usage of `pytorch_geometric` library. It was adapted from
-[protein_mincut_pool example](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/proteins_mincut_pool.py).
+[protein_topk_pool example](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/proteins_topk_pool.py).
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/README.md:4]:
```suggestion
the [protein_topk_pool example](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/proteins_topk_pool.py).
```

File: examples/graphs/proteins_pytorch_geometric/README.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 
 Key part of this example code is contained in the following functions in `model_def.py`:
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/README.md:6]:
```suggestion
The key parts of this example are contained in the following functions in `model_def.py`:
```

File: examples/graphs/proteins_pytorch_geometric/README.md [hunk 6]
```diff
@@ -42,7 +43,5 @@ const.yaml .`. The other configurations can be run by specifying the appropriate
 configuration file in place of `const.yaml`.
 
 ## Results
-The trial will achieve ~75% accuracy after training for 50 epochs, and will approach ~80% accuracy
-by epoch 300 in a few minutes of runtime on NVIDIA K80.
-
-NOTE: this example may have convergence issues with the default hyperparameters.
+The trial will achieve ~75% accuracy after training for 10 epochs, and will approach ~80% accuracy
+by epoch 200 in a few minutes of runtime on NVIDIA K80.
```
Comment (RehanSD) [inline examples/graphs/proteins_pytorch_geometric/README.md:47]:
```suggestion
by epoch 200 in a few minutes of runtime on an NVIDIA K80.
```

**Review by RehanSD — COMMENTED** [review]
Left a few comments - couple of quick questions:
1. Should we be putting adaptive and/or distributed in the nightlys?
2. Has this been tested in the nightlies on CircleCI? It doesn't seem to have been on the latest run, but wanted to check if you had done it on an earlier run.
Besides that, looks good to me!

File: examples/graphs/proteins_pytorch_geometric/distributed.yaml (hunk 3 above)
Comment (ioga) [reply examples/graphs/proteins_pytorch_geometric/distributed.yaml:21]: yes, it's mentioned in the README: https://github.com/determined-ai/determined/pull/2706/files#diff-5fe53f5cccc533a2935b718527bc4c347172512022f9628b372a30d89463ddf5R15

File: examples/graphs/proteins_pytorch_geometric/const.yaml (hunk 2 above)
Comment (ioga) [reply examples/graphs/proteins_pytorch_geometric/const.yaml:5]: fixed.

File: examples/graphs/proteins_pytorch_geometric/model_def.py (hunk 1 above)
Comment (ioga) [reply examples/graphs/proteins_pytorch_geometric/model_def.py:98]: it starts conflicting with our suggested type in the harness code (pytorch.TorchData), which makes things look very red if you have a type checker.

File: examples/graphs/proteins_pytorch_geometric/distributed.yaml (hunk 3 above)
Comment (RehanSD) [reply examples/graphs/proteins_pytorch_geometric/distributed.yaml:21]: Missed it, mb!

**Review by RehanSD — APPROVED** [review]
LGTM!

**Comment by ioga** [comment]
the test job has succeeded: 
https://app.circleci.com/pipelines/github/determined-ai/determined/14028/workflows/9293d070-afb3-4956-a658-c5b819c77bf0/jobs/416900/parallel-runs/0?filterBy=ALL
